[ARMv7] Remove TmpPair

justinmichaud opened the pull request:
#### 6c002efb9c38ad43297934b5b98d23c6c3f932cb
<pre>
[ARMv7] Remove TmpPair
<a href="https://bugs.webkit.org/show_bug.cgi?id=278548">https://bugs.webkit.org/show_bug.cgi?id=278548</a>

Reviewed by Keith Miller.

- Remove TmpPair.
- Extend Extract to support treating Int64s as Tuples

Instructions like add64 that require a full int64 now extract their
arugments from the Int64 input as if it were a tuple.

We also fix exceptions.

Canonical link: <a href="https://commits.webkit.org/283302@main">https://commits.webkit.org/283302@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/261abda0a009680b476d7bc0b043f762ea040e84

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65850 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45223 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18469 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/69879 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16459 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53022 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16741 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/52855 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11439 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/68917 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41746 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57002 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33487 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38419 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/14381 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15335 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/58964 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60261 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14725 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71582 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/65094 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9805 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14141 "Found 1 new test failure: http/wpt/mediastream/mediastreamtrackprocessor-videoframe-timestamp.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60169 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9837 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57069 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60455 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14521 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8102 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1742 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/86861 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41031 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15283 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42107 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43290 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41851 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on JSTests/wasm/wasm.json:
```diff
@@ -15,15 +15,15 @@
-        "funcref":       { "type": "varint7", "value":  -16, "b3type": "B3::Int64",  "width": 64 },
-        "externref":     { "type": "varint7", "value":  -17, "b3type": "B3::Int64",  "width": 64 },
+        "funcref":       { "type": "varint7", "value":  -16, "b3type": "(is32Bit() ? B3::Int32 : B3::Int64)",  "width": 64 },
+        "externref":     { "type": "varint7", "value":  -17, "b3type": "(is32Bit() ? B3::Int32 : B3::Int64)",  "width": 64 },
```
Can we just make these `B3::pointerType()`?

justinmichaud replied:
Ah, sweet!

kmiller68 reviewed (approved):
r=me.

justinmichaud committed:
[ARMv7] Remove TmpPair
https://bugs.webkit.org/show_bug.cgi?id=278548

Reviewed by Keith Miller.

- Remove TmpPair.
- Extend Extract to support treating Int64s as Tuples

Instructions like add64 that require a full int64 now extract their
arugments from the Int64 input as if it were a tuple.

We also fix exceptions.

Canonical link: https://commits.webkit.org/283302@main
